src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const arrLength = this.length;
+    let str = '';
+
+    for (let i = 0; i < arrLength; i++) {
+      if (typeof this[i] === 'undefined' || this[i] === null) {
```

Comment:
```suggestion
      if (this[i] === undefined || this[i] === null) {
```
typeof is redundant here